src/Raven.Server/ServerWide/Commands/Sharding/StartBucketMigrationCommand.cs

```diff
@@ -30,9 +32,15 @@ public StartBucketMigrationCommand(int bucket, int destShard, string database, s
             DestinationShard = destShard;
         }
 
+        public StartBucketMigrationCommand(int bucket, int sourceShard, int destShard, string database, string raftId, bool backgroundMigration) : this(bucket, destShard, database, raftId)
+        {
+            SourceShard = sourceShard;
+            BackgroundMigration = backgroundMigration;
```

Comment:
what is the reason to have `BackgroundMigration` here?

Reply:
because we don't want to change the `BucketRanges` of the database record (`DestinationMigrationConfirmCommand`) - so we need to separate the cases

Reply:
I expect it to be a noop so `BucketRanges` will not be changed. (otherwise it is a bug in the `MoveBucket` method)

Reply:
it throws in `ValidateBucketsMapping ` (`if (current.ShardNumber == lastShard)`, line 352)

Reply:
done